sdk/core/azure-core/src/main/java/com/azure/core/util/BinaryData.java

```diff
@@ -470,6 +470,35 @@ public static Mono<BinaryData> fromFlux(Flux<ByteBuffer> data, Long length, bool
         });
     }
 
+    /**
+     * Wraps the given flux in a BinaryData instance without subscribing to it.
+     *
+     * @param data Content to wrap.
+     * @param length Optional known length of the content in bytes.
+     * @return BinaryData instane wrapping the flux.
+     */
+    public static BinaryData wrapFlux(Flux<ByteBuffer> data, Long length) {
+        return wrapFlux(data, length, false);
+    }
+
+    /**
+     * Wraps the given flux in a BinaryData instance without subscribing to it.
+     *
+     * @param data Content to wrap.
+     * @param length Optional known length of the content in bytes.
+     * @param isReplayable Whether the content is replayable.
+     * @return BinaryData instane wrapping the flux.
+     */
+    public static BinaryData wrapFlux(Flux<ByteBuffer> data, Long length, boolean isReplayable) {
```

Comment:
The `fromFlux()` async variant takes a boolean called `bufferContent` and the sync variant takes a boolean called `isReplayable`. We should keep the two APIs consistent.

Reply:
This unfortunately seems to be a case where an enum should have been used instead of a boolean as there is more than two states, buffer now because my content is unknown, buffer eventually but not eagerly, don't buffer at all, possibly other cases I'm not thinking of at this time.

Reply:
@srnagar `bufferContent` can eagerly buffer data. This API cannot do that because it must be async to safely consume the Flux. To mirror, I imagine we would have to work off @alzimmermsft's comment, where the async API is redone with the enum, and this sync one mirror the enum overload, throwing an argument exception for eager buffering. Does that seem like an acceptable strategy?